op/oss.c

```diff
@@ -55,13 +55,47 @@ static int oss_reset(void)
 #ifndef AFMT_S32_BE
 #define AFMT_S32_BE	0x00002000
 #endif
-#ifndef AFMT_S24_PACKED
-#define AFMT_S24_PACKED	0x00040000
+#ifndef AFMT_S24_LE
+#define AFMT_S24_LE	0x00010000
+#endif
+#ifndef AFMT_S24_BE
+#define AFMT_S24_BE	0x00020000
+#endif
+#ifndef AFMT_U24_LE
+#define AFMT_U24_LE	0x00040000
```

Comment:
I was under the impression that the constants are the same across different OSS implementation, but that's apparently not true. S24_PACKED isn't even defined on FreeBSD and U24_LE isn't defined in OSSv4.

I don't really understand why the constants need to be added here at all to be honest.  I've amended the patch a little.